webrender/src/frame_builder.rs

```diff
@@ -1535,9 +1535,10 @@ impl<'a> LayerRectCalculationAndCullingPass<'a> {
             };
 
             if bounding_rect.is_none() {
-                bounding_rect =
-                    Some(clip_info.xf_rect.as_ref().map_or_else(DeviceIntRect::zero,
-                                                                |x| x.bounding_rect))
+                bounding_rect = Some(match clip_info.xf_rect {
```

Comment:
This all seems okay to me. A slightly cleaner change might be just delay the intersection until this point and write:
```rust
            if bounding_rect.is_none() {
                let rect = clip_info.xf_rect.as_ref().map_or_else(DeviceIntRect::zero,
                                                                  |x| x.bounding_rect);
                let rect = rect.intersection(self.screen_rect).unwrap_or_else(DeviceIntRect::zero);
                bounding_rect = Some(rect);
            }
```
I think that would be the only change necessary. This will also be useful if we ever need to use the real translated rectangle in the future again.